pkg/controllers/feature/feature_controller.go

```diff
@@ -462,6 +466,105 @@ func (r *frReconciler) updateFeatureSetStatus(ctx context.Context, fs *uiapi.Fea
 	return client.IgnoreNotFound(err)
 }
 
+func (r *frReconciler) calculateFeatureSetDependency(ctx context.Context) error {
+	enabled := pointer.Bool(r.feature.Status.Managed) && pointer.Bool(r.feature.Status.Enabled)
+
+	for _, name := range r.feature.Spec.Requirements.Features {
+		f := &uiapi.Feature{}
+		if err := r.client.Get(ctx, types.NamespacedName{Name: name}, f); err != nil {
+			if kerr.IsNotFound(err) {
+				// required feature isn't available yet
+				continue
+			}
+			return err
+		}
+
+		if !pointer.Bool(f.Status.Enabled) || f.Spec.FeatureSet == r.feature.Spec.FeatureSet {
+			// required feature isn't enabled or belongs to same feature set
+			continue
+		}
+
+		if err := r.updateFeatureSetDependencyStatus(ctx, f.Spec.FeatureSet, enabled); err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func (r *frReconciler) updateFeatureSetDependencyStatus(ctx context.Context, fsName string, enabled bool) error {
+	fs := &uiapi.FeatureSet{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: fsName}, fs)
+	if err != nil {
+		return err
+	}
```

Comment:
We can move these dependancy related new changes to a seperate file.

Reply:
sounds good too

Reply:
moved